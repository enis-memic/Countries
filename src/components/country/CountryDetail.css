.country-detail-container {
  max-width: 80%;
  height: 100vh;
  color: white;
  margin: 0 auto;
}

.country-detail-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.country-detail-content h1 {
  margin-bottom: 2rem;
}

.country-detail-image {
  width: 40%;
  padding-top: 2rem;
}

.country-detail-right {
  width: 100%;
}

.details {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.detail-left {
  min-width: 50%;
}

.detail-left p {
  margin-bottom: 1rem;
  font-weight: 600;
}

.detail-left span {
  font-weight: 300;
  margin-left: 4px;
}

.detail-right {
  min-width: 100%;
}

.detail-right p {
  margin-bottom: 1rem;
  font-weight: 600;
}

.detail-right span {
  font-weight: 300;
  margin-left: 4px;
}

@media (min-width: 480px) {
  .country-detail-content {
    display: grid;
    grid-template-columns: 1fr 1fr;
    justify-content: space-around;
    margin-top: 10rem;
  }

  .country-detail-image {
    width: 70%;
  }

  h1 {
    font-size: 50px;
  }

  .details p {
    font-size: 30px;
  }
}
