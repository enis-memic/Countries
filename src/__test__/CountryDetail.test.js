import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CountryDetail from '../components/country/CountryDetail';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('CountryDetail', () => {
  let store;
  const initialState = {
    country: {
      error: false,
      loading: false,
      success: true,
      countrySearched: [
        {
          name: { common: 'Country Name' },
          population: 1000000,
          region: 'Test Region',
          capital: 'Test Capital',
          tld: ['.tld'],
          timezones: ['UTC'],
          languages: {
            eng: 'English',
            spa: 'Spanish',
          },
          coatOfArms: {
            png: 'https://test.com/test.png',
            alt: 'Test Alt',
          },
        },
      ],
    },
  };

  beforeEach(() => {
    store = mockStore(initialState);
  });

  test('renders CountryDetail component', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/code']}>
          <CountryDetail />
        </MemoryRouter>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
