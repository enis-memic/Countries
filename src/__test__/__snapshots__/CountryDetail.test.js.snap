// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetail renders CountryDetail component 1`] = `
<section
  class="country-detail-container"
>
  <div
    class="country-detail-content"
  >
    <img
      alt="Test Alt"
      class="country-detail-image"
      src="https://test.com/test.png"
    />
    <div
      class="country-detail-right"
    >
      <h1>
        Country Name
      </h1>
      <div
        class="details"
      >
        <div
          class="detail-left"
        >
          <p>
            Offcial Name:
             
            <span />
          </p>
          <p>
            Population:
             
            <span>
              1000000
            </span>
          </p>
          <p>
            Region:
             
            <span>
              Test Region
            </span>
          </p>
          <p>
            Time Zone:
             
            <span>
              UTC
            </span>
          </p>
          <p>
            Capital:
             
            <span>
              Test Capital
            </span>
          </p>
        </div>
        <div
          class="detail-right"
        >
          <p>
            Top Level Domain:
             
            <span>
              .tld
            </span>
          </p>
          <p>
            Languages:
            <span>
              English,Spanish
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
